soc/xtensa/nxp_adsp/common/soc.c

```diff
@@ -20,6 +20,10 @@
 #include <zephyr/logging/log.h>
 LOG_MODULE_REGISTER(soc);
 
+
+/* temporary workaround the fact that not all i.MX8 SoCs use IRQ_STEER driver */
+#ifndef CONFIG_NXP_IRQSTEER
```

Comment:
Can't we just disable the dts node until everything is ready?

Reply:
Depends. If i.MX8ULP uses IRQSTEER this would mean having to hold off all other i.MX8 boards (3 of them) until i.MX8ULP is ready (meaning HAL support as well), which may or may not be ok. If not, then I can just enable IRQ_STEER on i.MX8QM and QXP in this PR and solve this.

Reply:
ACK, thanks! Will also enable IRQ_STEER for QM and QXP in this patch. In the meantime, I'll have to check if ULP still requires the definitions from `soc.c` and decide what to do with it.

Reply:
Update: removed commit which enables the IRQ_STEER driver on i.MX8MP